File: cafeoccidente-backend/src/main/java/com/cafeoccidente/backend/users/service/UserService.java
package com.cafeoccidente.backend.users.service;

import com.cafeoccidente.backend.users.dto.UserRequest;
import com.cafeoccidente.backend.users.dto.UserResponse;
import java.util.List;

public interface UserService {

    List<UserResponse> list();

    UserResponse create(UserRequest request);
}

File: cafe-occidente-frontend/src/app/core/services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { API_BASE_URL } from '../config/api.config';
import { CreateUserRequest, User } from '../models/user.model';

/** Unica responsabilidad: llamadas HTTP al backend para el recurso User. */
@Injectable({ providedIn: 'root' })
export class UserService {
  private readonly http = inject(HttpClient);

  list(): Observable<User[]> {
    return this.http.get<User[]>(`${API_BASE_URL}/users`);
  }

  create(request: CreateUserRequest): Observable<User> {
    return this.http.post<User>(`${API_BASE_URL}/users`, request);
  }
}
